### ISSUE TITLE
RandomMatrix - name clash with other packages

### ISSUE THREAD

**Issue by alex-konovalov** [gap-packages/Semigroups#352]
Semigroups now defines `RandomMatrix` as a function. However, some other packages (namely GradedModules and MatricesForHomalg, already in GAP 4.8.7 release) define it as operation. Therefore, if any of those packages is loaded, then it's not possible to load Semigroups: 
```
#I  Semigroups: start reading file 'init.g'
Error, BIND_GLOBAL: variable `RandomMatrix' must be unbound called from
BIND_GLOBAL( name, NEW_OPERATION_ARGS( name ) 
 ); at /data/gap-jenkins/workspace/GAP-pkg-update-stable-quicktest/GAPCOPTS/64\
build/GAPGMP/gmp/GAPTARGET/packagesload/label/kovacs/GAP-pkg-update-stable-sna\
pshot/lib/oper.g:1686 called from
<function "DeclareGlobalFunction">( <arguments> )
 called from read-eval loop at line 
77
  of /data/gap-jenkins/workspace/GAP-pkg-update-stable-quicktest/GAPCOPTS/64bu\
ild/GAPGMP/gmp/GAPTARGET/packagesload/label/kovacs/GAP-pkg-update-stable-snaps\
hot/pkg/semigroups-3.0.4/gap/elements/semiringmat.gd
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk> 
```
I think this could have been discovered earlier by running `make testinstall` which will run testinstall twice, with only needed packages and with loading all packages.

Would you be able to fix this and make a new release of Semigroups, please?

**Comment by james-d-mitchell** [comment]
Yes sure no problem. But Semigroups 3.0.? requires GAP 4.9 so should not be included in an earlier release.

**Comment by alex-konovalov** [comment]
Thanks - specified that in https://github.com/gap-system/gap-distribution/commit/b474df19999f5b39c63df5767b326deca9def70b (what this file is doing is documented in https://github.com/gap-system/gap-distribution/blob/master/DistributionUpdate/PackageUpdate/README.md)

**Comment by alex-konovalov** [comment]
For now, I've temporary disabled Semigroups for tests in master branch in https://github.com/gap-system/gap-distribution/commit/b0f1f90b3c027eba5d4d5713522d7e6337a68276

**Comment by wilfwilson** [comment]
Resolved by PR #359.
